tools/extensions/phpstorm/Civi/PhpStorm/PhpStormMetadata.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Civi\PhpStorm;
+
+/**
+ * Utility class for building new metadata files. For example:
+ *
+ * $builder = new PhpStormMetadata('foobar', __CLASS__);
+ * $builder->addOverrideMap('\Foo::bar()', ['gee' => '\Whiz\Bang']);
+ * $builder->write();
+ *
+ * @link https://www.jetbrains.com/help/phpstorm/2021.3/ide-advanced-metadata.html
+ */
+class PhpStormMetadata {
+
+  /**
+   * Short symbolic name. Maps to filename.
+   *
+   * @var string
+   */
+  protected $name;
+
+  /**
+   * The party responsible for building this thing.
+   *
+   * @var string
+   */
+  protected $attribution;
+
+  /**
+   * @var string
+   */
+  protected $buffer;
+
+  /**
+   * @param string $name
+   *   Short symbolic name. This will be used in the filename.
+   * @param string $attribution
+   *   The party responsible for running this builder. If someone finds buggy, where should they look first?
+   */
+  public function __construct(string $name, string $attribution) {
+    $this->name = $name;
+    $this->attribution = $attribution;
+    $this->buffer = '';
+  }
+
+  /**
+   * @param string $for
+   * @param array $map
+   *   Array(string $parameterId => string $className)
+   *
+   * @return $this
+   */
+  public function addOverrideMap(string $for, array $map) {
+    // PhpStorm 2022.3.1: 'Civi\\Foo' doesn't work, but 'Civi\Foo' does.
+    $mapData = str_replace('\\\\', '\\', var_export($map, 1));
+    $this->buffer .= "override($for, map(\n$mapData);\n";
```

Comment:
@totten this is missing a closing `)`. PhpStorm doesn't complain because the metadata file never actually needs to run, but it would be a parse error otherwise.

Reply:
Ooops. Thanks @colemanw. Fixed.